visual-java/src/main/java/com/saucelabs/visual/VisualApi.java

```diff
@@ -218,7 +236,13 @@ private WebdriverSessionInfoQuery.Result webdriverSessionInfo() {
     WebdriverSessionInfoQuery query =
         new WebdriverSessionInfoQuery(
             new WebdriverSessionInfoQuery.WebdriverSessionInfoIn(this.jobId, this.sessionId));
-    return this.client.execute(query, WebdriverSessionInfoQuery.Data.class).result;
+    WebdriverSessionInfoQuery.Data response =
+        this.client.execute(query, WebdriverSessionInfoQuery.Data.class);
+    if (response.result == null) {
+      throw new VisualApiException(
```

Comment:
Can we also log the `traceId` there ?

It will be helpful to troubleshoot customers